coalib/io/FileFactory.py

```diff
@@ -0,0 +1,115 @@
+import linecache
+import os
+
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('_filename', '_stamp')
+class FileFactory:
+    """
+    The FileFactory is a wrapper object to replace the actual file contents
+    from the file-dict. The object contains the following information about
+    the file:
+
+        * The filename (absolute file path).
+        * The parent directory the file is located in.
+        * The modification timestamp of the file.
+        * The file in the form of a string.
+        * The file in the form of a list.
+        * The file in binary format.
+
+    The equality of the object is checked just based on its filename and
+    timestamp.
+
+    To initialize a FileFactory object for a file:
+
+    >>> fp = FileFactory("./tests/io/FileFactoryTestFiles/test.txt")
+
+    Make the object into an iterable when required
+
+    >>> for line in fp:
+    ...     print(line)
+    This is a test file.
+    <BLANKLINE>
+
+    Retrieve a single line:
+
+    >>> fp.line(1)
```

Comment:
Are line indices starting with 1 ? If so, why ?

Reply:
I think its because of `linecache`'s it counts lines from 1

Reply:
1. Choose a convention that you think makes sense. coala's `Result`s have a one-based convention, but usually lists, tuples, array-like-things are zero-based due to technical reasons (e.g. see for-loops).
2. Document it! :) This is extremely important for the user to know.

My feeling: I tend to zero-based notation.